paprika/setup.py

```diff
@@ -4,759 +4,723 @@
-                                   dummy_atom_tuples=[(0, 0, -6.0),
-                                                      (0, 0, -9.0),
-                                                      (0, 2.2, -11.2)])
-        except FileNotFoundError:
-            logger.warning(f"Missing {solvated_pdb}")
+        return restraints
 
+    @classmethod
+    def build_symmetry_restraints(
+        cls,
+        coordinate_path: str,
+        n_attach_windows: int,
+        restraint_schemas: List[Dict[str, Any]],
+        use_amber_indices: bool = False,
+    ) -> List[DAT_restraint]:
+        """A method to convert a set of symmetry restraints defined by their 'schemas'
+        into corresponding ``DAT_restraint``objects.
 
-        self._wrap(dummy_pdb)
+        Each 'schema' should be a dictionary with:
 
-        # Add dummy atoms to System
-        if solvated_xml is not None:
+            * an ``atoms`` entry with a value of the atom selection make which specifies
+              which atoms the restraint will apply to
+            * a ``force_constant`` entry which specifies the force constant of the
+              restraint.
 
-            try:
+        These 'schemas` map directly to the 'restraints -> symmetry_correction
+        -> restraint' dictionaries specified in the `taproom` guest YAML files.
 
-                system = read_openmm_system_from_xml(solvated_xml)
-                system = self._add_dummy_to_System(system, dummy_atom_tuples=[(0, 0, -6.0),
-                                                                              (0, 0, -9.0),
-                                                                              (0, 2.2, -11.2)])
-                system_xml = openmm.XmlSerializer.serialize(system)
-                with open(dummy_xml, "w") as file:
-                    file.write(system_xml)
+        Parameters
+        ----------
+        coordinate_path
+            The path to the coordinate file which the restraints will be applied to.
+            This should contain either the host or the complex, the dummy atoms and
+            and solvent.
+        n_attach_windows
+            The total number of attach windows being used in the APR calculation.
+        restraint_schemas
+            The list of dictionaries which provide the settings to use for each
+            symmetry restraint to add.
+        use_amber_indices
+            Whether to use amber based (i.e. starting from 1) restraint indices or
+            OpenMM based (i.e. starting from 0) indices.
 
-            except FileNotFoundError:
-                logger.warning(f"Missing {solvated_xml}")
+        Returns
+        -------
+            The constructed symmetry restraint objects.
+        """
 
-    @staticmethod
-    def _wrap(file, mask=":DM3"):
-        logging.info(f"Re-wrapping {file} to avoid pulling near periodic boundaries.")
-        structure = pmd.load_file(file, structure=True)
+        restraints = []
```

Comment:
I'm not sure if it's just the `diff` rendering, but it looks like you removed the `_wrap` function. Might it be useful to keep that somewhere handy, in case we need to use it again?

<img width="787" alt="image" src="https://user-images.githubusercontent.com/5393995/90526965-ddd53b00-e125-11ea-9bdc-a9b743d0beb1.png">